src/pages/SignUp.js

```diff
@@ -1,7 +1,111 @@
+import { Link, useHistory } from "react-router-dom";
+import { useState } from "react";
+import { API } from "../service/auth";
+import { EnterContainer, LogoHolder, LoginForm, EnterButton, EnterInput } from "../SharedStyles/LogInSignUp";
+
 function SignUp() {
-    return (
-        <div>SignUp</div>
-    )
+
+  const history = useHistory();
+  const [createEmail, setCreateEmail] = useState("");
+  const [createPassword, setCreatePassword] = useState("");
+  const [createUsername, setCreateUsername] = useState("");
+  const [createImage, setCreateImage] = useState("");
+  const [enabled, setEnabled] = useState(true);
+
+  function signUpSuccess(response) {
+    if (response.status === 200 || response.status === 201) {
+      alert(
+        "You account has been created! Now you only need to log in to start having fun! :D"
+      );
+      setEnabled(true);
+      setCreateEmail("");
+      setCreatePassword("");
+      setCreateUsername("");
+      setCreateImage("");
+      history.push("/");
+    }
+  }
+
+  function signUpFailure(response) {
+    if (response.response.status === 403) {
+      alert("This e-mail has already been used to create an account. Please, log in or try to use another e-mail address.");
+    } else {
+      alert("Something went wrong. Please, check the fields and try again.");
+    }
+    setEnabled(true);
+  }
+
+  function createAccount (e) {
+    e.preventDefault();
+    setEnabled(false);
+
+    const body = {
+      "email": createEmail,
+      "password": createPassword,
+      "username": createUsername,
+      "pictureUrl": createImage
+    }
```

Comment:
Não é um problema deixar assim, mas só para dizer que não é obrigatório colocar a chave de um objeto entre aspas quando for enviar uma requisição. O próprio axios transforma o objeto em JSON ;)

Reply:
Ah, achei que ia ficar mais seguro pra ir certinho, mas já alterei aqui também. :)